### DISCUSSION TITLE
Permissions with creating a poll on my channel

### DISCUSSION THREAD

**Discussion by discrait** [TwitchIO/TwitchIO#389] [Questions] [answered]
I am trying to create a python script that creates a poll in my twitch channel and after it is done it should print the results. However I can't even get the first step to work, which is creating a poll. The code below is what I have right now. (sensitive data censored with #########)

```
import os
import requests
import json
from twitchio import Client
from twitchio.ext import commands
from twitchio import PartialUser
from datetime import datetime, timedelta

# Set your Twitch API credentials
channel_id = '#########'
oauth_token = '#########'
client_id = '#########'

bot = commands.Bot(
    token=oauth_token,
    client_id=client_id,
    nick='BOT',
    client_secret="#########",
    prefix='!',
    initial_channels=["#channel_1#"]
)

@bot.event()
async def event_ready():
    channel = bot.get_channel("#channel_1#")
    await channel.send("this works perfectly fine")
    
    user = bot.create_user(user_id=#########, user_name="#channel_1#")
    await user.create_poll(token="#########",title="test poll",choices=["1","2","3"],duration=60)

    
if __name__ == "__main__":
    bot.run()
```
After running this it prints this:
```
Unauthorized {'error': 'Unauthorized', 'status': 401, 'message': 'Missing scope: channel:manage:polls'}
```
and then gives me the error:
```
raise errors.Unauthorized("You're not authorized to use this route.")
twitchio.errors.Unauthorized: You're not authorized to use this route.
```
I can't find anything about how to change authorization in the documentation or anyone online with the same problem.
The only thing I could find was ChatGPT telling me "The error message is stating that you are missing the channel:manage:polls scope in your Twitch API credentials. You need to generate a new OAuth token with this scope added in order to create polls for your channel."

However I can't find anything about changing scope in my twitch dev console.

Does anyone know how to tackle this? or if I'm tackling this process of creating a poll completely wrong?

Kind regards and thanks in advance

**Comment by discrait** [comment, marked as answer]
SOLVED!

Twitchio works with something called scopes for their tokens so if your token doesnt have the correct scopes (authorizations) its of course not going to allow you to do something.

Its an added level of security.

To get a token with the right authorizations check: twitchtokengenerator.com for a custom scope token :)

**Reply by chillymosh** [reply]
Ideally you would only use TTG for testing purposes and would use your own webserver for oauth flow.
Otherwise you are going to have a bad time regarding refreshes or any breaking changes TTG may make in the future.

`nick` is also not used for Bot so you can omit that.
